packages/strapi-plugin-content-type-builder/admin/src/containers/ModelPage/index.js

```diff
@@ -1,320 +1,662 @@
+    return !isEqual(initialData[currentModel], modifiedData[currentModel]) && this.getSearch() === '';
+  };
 
-    return (
-      <li className={linkStyles.pluginLeftMenuLink}>
-        <NavLink className={linkStyles.link} to={`/plugins/content-type-builder/models/${props.link.name}${props.link.source ? `&source=${props.link.source}` : ''}`} activeClassName={linkStyles.linkActive}>
-          <div>
-            <i className={`fa fa-caret-square-o-right`} />
-          </div>
-          <div className={styles.contentContainer}>
+  isUpdatingTemporaryContentType = (modelName = this.getModelName()) => {
+    const { models } = this.props;
+    /* istanbul ignore next */
+    const currentModel = models.find(model => model.name === modelName) || { isTemporary: true };
 
-            <span className={spanStyle}>{startCase(props.link.name)}</span>
-            <span style={{ marginLeft: '1rem', fontStyle: 'italic' }}>{temporary}{pluginSource}</span>
-          </div>
-        </NavLink>
-      </li>
-    );
-  }
+    const { isTemporary } = currentModel;
 
-  renderListTitle = (props, listStyles) => {
-    const availableNumber = size(props.listContent.attributes);
-    const title = availableNumber > 1 ? 'content-type-builder.modelPage.contentType.list.title.plural'
-      : 'content-type-builder.modelPage.contentType.list.title.singular';
+    return isTemporary;
+  };
 
-    const relationShipNumber = props.listContent.attributes.filter(attr => has(attr.params, 'target')).length;
+  setPrompt = () => this.setState({ removePrompt: false });
 
-    const relationShipTitle = relationShipNumber > 1 ? 'content-type-builder.modelPage.contentType.list.relationShipTitle.plural'
-      : 'content-type-builder.modelPage.contentType.list.relationShipTitle.singular';
+  isTryingToEditAnUnknownAttribute = () => {
+    const hasAttribute = Object.keys(this.getModelAttributes()).indexOf(this.getAttributeName()) !== -1;
 
-    let fullTitle;
+    return this.getActionType() === 'edit' && this.getModalType() === 'attributeForm' && !hasAttribute;
+  };
 
-    if (relationShipNumber > 0) {
-      fullTitle = (
-        <div className={listStyles.titleContainer}>
-          {availableNumber} <FormattedMessage id={title} /> <FormattedMessage id={'content-type-builder.modelPage.contentType.list.title.including'} /> {relationShipNumber} <FormattedMessage id={relationShipTitle} />
-        </div>
-      );
-    } else {
-      fullTitle = (
-        <div className={listStyles.titleContainer}>
-          {availableNumber} <FormattedMessage id={title} />
+  shouldRedirect = () => {
+    const { models } = this.props;
 
-        </div>
+    return (
+      models.findIndex(model => model.name === this.getModelName()) === -1 ||
+      this.isTryingToEditAnUnknownAttribute()
+    );
+  };
+
+  toggleModalWarning = () => this.setState(prevState => ({ showWarning: !prevState.showWarning }));
+
+  wait = async () => {
```

Comment:
What is it for?

Reply:
Well, I have added a prompt so the user if notified when he tries to navigate to another CT without saving his modifications. Basically the prompt is sensible to the URL change so when we open the modal the prompt can get displayed because the URL changes with the search and we don't want this behaviour since the user is not leaving the edited CT.

The bug is due to our nested routers so it's really difficult to prevent this behaviour... The only way I found to solve this without making some significant updates in the strapi-admin is to play with a state so the prompts is deactivated just before changing the URL to open the modal...

Reply:
I hope my explanation makes sense.